contracts/StablecoinConverter.sol

```diff
@@ -133,9 +133,11 @@ contract StablecoinConverter is EpochTokenLocker {
             batchIndex == getCurrentStateIndex() - 1,
             "Solutions are no longer accepted for this batch"
         );
+        require(checkPriceOrdering(tokenIdsForPrice), "prices are not ordered by tokenId");
         undoPreviousSolution(batchIndex);
         updateCurrentPrices(prices, tokenIdsForPrice);
         delete previousSolutionTrades;
+        int[] memory tokenConservation = new int[](prices.length);
```

Comment:
It's very annoying that there are no memory mappings. There might be a more appropriate data-structure to use here (e.g. a sorted binary tree) that can be abstracted away via a library, but I think this is fine for now.